src/managers/backup.py

```diff
@@ -176,6 +183,55 @@ def format_backups_table(
 
         return out_f.getvalue()
 
+    def is_snapshot_in_bucket(self, backup_id: str) -> bool:
+        """Check whether the requested snapshot to restore is in the object storage."""
+        try:
+            bucket = self.bucket
+        except KeyError:
+            return False
+
+        try:
+            content = bucket.meta.client.head_object(
+                Bucket=bucket.name, Key=os.path.join(self.backups_path, backup_id, "snapshot")
+            )
+        except ClientError as ex:
+            if "(404)" in ex.args[0]:
+                return False
+            raise
+        return content.get("ResponseMetadata", None) is not None
+
+    def restore_snapshot(self, backup_id: str, workload: ZKWorkload) -> None:
+        """Download and restore a snapshot.
+
+        Restoring requires removing the previous files on disk. In this step, we move them to a backup folder,
+        so that we can manually restore them if something goes wrong.
+        """
+        data_dir = Path(PATHS["DATA"]) / "data" / "version-2"
+        data_log_dir = Path(PATHS["DATA"]) / "data-log" / "version-2"
+
+        workload.exec(["bash", "-c", f"cp -rp {data_dir} {data_dir.parent / 'version-2.bak'}"])
+        workload.exec(
+            ["bash", "-c", f"cp -rp {data_log_dir} {data_log_dir.parent / 'version-2.bak'}"]
+        )
+
+        workload.exec(["bash", "-c", f"rm -rf {data_dir / '*'}"])
+        workload.exec(["bash", "-c", f"rm -rf {data_log_dir / '*'}"])
+
+        restored_snapshot = data_dir / "snapshot.0"
```

Comment:
**thought**: Now that I am writing the documentation and adding a link to the ZK docs about the zxid, I realise this might be weird to our users that they restore a snapshot with a zxid from epoch 5 for instance, and end up with a deployment on epoch 1 once everything settles down.
Instead of `snapshot.0`, maybe we should use the zxid value we have